### ISSUE TITLE
runtime error: invalid memory address or nil  pointer dereference

### ISSUE THREAD

**Issue by UROjQ6r80p** [markus-wa/demoinfocs-golang#461]
**Describe the bug**
```
2023/10/27 13:24:27 runtime error: invalid memory address or nil 
pointer dereference
stacktrace:
goroutine 6 [running]:
runtime/debug.Stack()
        C:/Program Files/Go/src/runtime/debug/stack.go:24 +0x5e  
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.NewParserWithConfig.func1({0x1322ee0, 0xc000a9f2c0})
```

**To Reproduce**
Match: https://www.hltv.org/matches/2367504/monte-vs-og-roobet-cup-2023
file: monte-vs-og-m1-mirage-p2.dem

Code:
```
package main

import (
	"fmt"
	"log"
	"os"
	dem "github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs"
	common "github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs/common"
)

func main() {
	f, err := os.Open("./demos/monte-vs-og-m1-mirage-p2.dem")
	if err != nil {
		log.Panic(err)
	}
	defer f.Close()

	p := dem.NewParser(f)
	defer p.Close()

	p.ParseHeader()

	err = p.ParseToEnd()
	if err != nil {
		log.Panic(err)
	}

	fmt.Println(p.Header().MapName)
}
```

**Expected behavior**
No error.

**Library version**
github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5

**Additional context**
Windows 10

**Comment by markus-wa** [comment]
That stacktrace looks very short - is that the full one? if not can you post the full stack trace? (I can't test it myself at the moment)

**Comment by UROjQ6r80p** [comment]
> That stacktrace looks very short - is that the full one? if not can you post the full stack trace? (I can't test it myself at the moment)
Sorry, here is the full one:

```
2023/11/01 13:18:29 runtime error: invalid memory address or nil 
pointer dereference
stacktrace:
goroutine 6 [running]:
runtime/debug.Stack()
        C:/Program Files/Go/src/runtime/debug/stack.go:24 +0x5e  
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.NewParserWithConfig.func1({0x12f2ee0, 0xc0009d6b00})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5/pkg/demoinfocs/parser.go:388 +0x2a     
github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover.func1()
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:197 +0x3b
panic({0x12f2ee0?, 0xc0009d6b00?})
        C:/Program Files/Go/src/runtime/panic.go:914 +0x21f      
github.com/markus-wa/godispatch.callConsumerCode.func1()
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:122 +0x54
panic({0x12c34e0?, 0x17f9f60?})
        C:/Program Files/Go/src/runtime/panic.go:914 +0x21f      
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.(*parser).bindBomb.func2({0x140dc20, 0xc0004a7880})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5/pkg/demoinfocs/datatables.go:165 +0x2bcgithub.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs/sendtables2.(*Parser).OnPacketEntities(0xc00007e140, 0xc0006ed110)       
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5/pkg/demoinfocs/sendtables2/entity.go:495 +0xa7c
reflect.Value.call({0x12a1ba0?, 0xc0000084b0?, 0xc00029ede0?}, {0x13a5369, 0x4}, {0xc00045bef8, 0x1, 0x1?})
        C:/Program Files/Go/src/reflect/value.go:596 +0xce7      
reflect.Value.Call({0x12a1ba0?, 0xc0000084b0?, 0x0?}, {0xc00045bef8?, 0x4700000000000000?, 0x2?})
        C:/Program Files/Go/src/reflect/value.go:380 +0xb9       
github.com/markus-wa/godispatch.callConsumerCode({0x12a1ba0?, 0xc0000084b0?, 0x1386f40?}, {0xc00045bef8?, 0xf770a6?, 0xf9d82e?})  
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:125 +0x3f
github.com/markus-wa/godispatch.(*Dispatcher).Dispatch(0xc00007c120, {0x1386f40, 0xc0006ed110})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:111 +0x1d4
github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover(0xc00007c138?, {0x1386f40?, 0xc0006ed110?})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:202 +0x4f
github.com/markus-wa/godispatch.(*Dispatcher).dispatchQueue(0xc00007c120, 0x0?)
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:177 +0xd0
created by github.com/markus-wa/godispatch.(*Dispatcher).AddQueues in goroutine 1
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:162 +0x18a
panic: runtime error: invalid memory address or nil pointer dereference
stacktrace:
goroutine 6 [running]:
runtime/debug.Stack()
        C:/Program Files/Go/src/runtime/debug/stack.go:24 +0x5e  
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.NewParserWithConfig.func1({0x12f2ee0, 0xc0009d6b00})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5/pkg/demoinfocs/parser.go:388 +0x2a     
github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover.func1()
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:197 +0x3b
panic({0x12f2ee0?, 0xc0009d6b00?})
        C:/Program Files/Go/src/runtime/panic.go:914 +0x21f      
github.com/markus-wa/godispatch.callConsumerCode.func1()
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:122 +0x54
panic({0x12c34e0?, 0x17f9f60?})
        C:/Program Files/Go/src/runtime/panic.go:914 +0x21f      
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.(*parser).bindBomb.func2({0x140dc20, 0xc0004a7880})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5/pkg/demoinfocs/datatables.go:165 +0x2bcgithub.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs/sendtables2.(*Parser).OnPacketEntities(0xc00007e140, 0xc0006ed110)       
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.5/pkg/demoinfocs/sendtables2/entity.go:495 +0xa7c
reflect.Value.call({0x12a1ba0?, 0xc0000084b0?, 0xc00029ede0?}, {0x13a5369, 0x4}, {0xc00045bef8, 0x1, 0x1?})
        C:/Program Files/Go/src/reflect/value.go:596 +0xce7      
reflect.Value.Call({0x12a1ba0?, 0xc0000084b0?, 0x0?}, {0xc00045bef8?, 0x4700000000000000?, 0x2?})
        C:/Program Files/Go/src/reflect/value.go:380 +0xb9       
github.com/markus-wa/godispatch.callConsumerCode({0x12a1ba0?, 0xc0000084b0?, 0x1386f40?}, {0xc00045bef8?, 0xf770a6?, 0xf9d82e?})  
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:125 +0x3f
github.com/markus-wa/godispatch.(*Dispatcher).Dispatch(0xc00007c120, {0x1386f40, 0xc0006ed110})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:111 +0x1d4
github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover(0xc00007c138?, {0x1386f40?, 0xc0006ed110?})
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:202 +0x4f
github.com/markus-wa/godispatch.(*Dispatcher).dispatchQueue(0xc00007c120, 0x0?)
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:177 +0xd0
created by github.com/markus-wa/godispatch.(*Dispatcher).AddQueues in goroutine 1
        C:/Users/Maciej/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:162 +0x18a


goroutine 1 [running]:
log.Panic({0xc000489e78?, 0x7?, 0x0?})
        C:/Program Files/Go/src/log/log.go:432 +0x5a
main.main()
        C:/Users/Maciej/Desktop/___LAB___/cs2-demo-parser/golang/test.go:25 +0x1bf
exit status 2
```

**Comment by markus-wa** [comment]
thanks! fixed by @akiver in #464
